experiments/modern/src/utils.js

```diff
@@ -96,3 +96,15 @@ export async function inlineIncludes(xml, zip) {
     return includedFile.children;
   });
 }
+
+// Operations on trees
+export function findParentNodeOfType(node, type) {
+  let n = node;
+  while(n.parent !== null) {
+    n = n.parent;
+    if ((!Array.isArray(type) && n.xmlNode.name === type) ||
+        (Array.isArray(type) && type.includes(n.xmlNode.name))) {
```

Comment:
Rather than checking the XML node name (string) can we pass an array of classes and do an instanceof check here?

Maybe that fails with inheritance?

Reply:
yea, I think that would be better. Same question about do we need to import all those classes just to check? And would need to check inheritance, but would like to think it would work hah